contracts/src/Teleporter/upgrades/TeleporterUpgradeable.sol

```diff
@@ -217,6 +221,31 @@ abstract contract TeleporterUpgradeable is Context, ITeleporterReceiver, Reentra
      */
     function _checkTeleporterUpgradeAccess() internal virtual;
 
+    /**
+     * @dev Sends a cross chain message using the TeleporterMessenger contract.
+     * Also increases the Teleporter contract's allowance for the fee token if the fee amount is non-zero.
+     *
+     * @return `messageID` The unique identifier for the Teleporter message.
+     */
+    function _sendTeleporterMessage(TeleporterMessageInput memory messageInput)
+        internal
+        virtual
+        returns (bytes32)
+    {
+        ITeleporterMessenger teleporterMessenger = _getTeleporterMessenger();
+        // For non-zero fee amounts increase the Teleporter contract's allowance.
+        if (messageInput.feeInfo.amount > 0) {
+            require(
+                messageInput.feeInfo.feeTokenAddress != address(0),
+                "TeleporterUpgradeable: zero fee token address"
+            );
+            IERC20(messageInput.feeInfo.feeTokenAddress).safeIncreaseAllowance(
+                address(teleporterMessenger), messageInput.feeInfo.amount
+            );
+        }
```

Comment:
I think having the call to `safeTransferFrom` outside of this function and `safeIncreaseAllowance` is a little strange, but I understand why we've done things this way and think it's the right choice. Can we leave comments here, and anywhere else that makes sense that note that fees should transferred to the contract before calling `_sendTeleporterMessage`?